### PR TITLE
server: make sure the API server boots first

### PR DESCRIPTION
**Pull request by nicks**
Hello @milas,

Please review the following commits I made in branch nicks/setup:

6b39af73253c133d7008e599d197d1dd8a8b7620 (2021-02-23 12:28:19 -0500)
server: make sure the API server boots first

f996027564f93d1694546cfc2dcd3aece498de49 (2021-02-23 12:19:02 -0500)
lint: run goimports, remove dead code

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/hud/server/controller.go [hunk 1]
```diff
@@ -54,31 +53,30 @@ func (s *HeadsUpServerController) TearDown(ctx context.Context) {
...
 	}
+	return nil
 }
 
-// Merge the APIServer and the Tilt Web server into a single handler,
-// and attach them both to the public listener.
-//
-// TODO(nick): We could move this to SetUp if SetUp had error-handling.
-func (s *HeadsUpServerController) serve(ctx context.Context, st store.RStore) error {
+func (s *HeadsUpServerController) setUpHelper(ctx context.Context, st store.RStore) error {
```
Comment (milas) [inline internal/hud/server/controller.go:79]: Super unimportant, but this fn is so tiny now it could probably just get merged into `SetUp` at this point

**Review by milas — APPROVED** [review]
(Apologies for delay, I must have gotten distracted while submitting review yesterday)

Changes look fine to me - it might be worth changing the various `_ = st.AddSubscriber(...)` calls throughout tests to be `require.NoError(t, st.AddSubscriber(...))` to future proof them a bit (obviously none of the existing fns return errors currently, but might mean some hard to debug test failures in the future if that changes).

File: internal/hud/server/controller.go (hunk 1 above)
Comment (nicks) [reply internal/hud/server/controller.go:79]: ya, it's more for sharing the error-handling/cancel

**Comment by nicks** [comment]
updated the AddSubscriber calls!

**Pull request merged**